process/notifier/sovereignNotifier.go

```diff
@@ -150,21 +150,46 @@ func (notifier *sovereignNotifier) createIncomingMbsFromTxs(txs map[string]*outp
 
 		log.Info("found incoming tx", "tx hash", txHash, "receiver", hex.EncodeToString(receiver))
 
-		txHashes = append(txHashes, hashBytes)
 		execOrderTxHashMap[string(hashBytes)] = tx.GetExecutionOrder()
+
+		sender := tx.GetTransaction().GetSndAddr()
+		senderShardID := notifier.shardCoordinator.ComputeId(sender)
+		shardIDTxHashMap[senderShardID] = append(shardIDTxHashMap[senderShardID], hashBytes)
```

Comment:
I think L146-L149 could be removed and use directly txHash in L153 instead string(hashBytes) and here you can use []byte(txHash) instead hashBytes as well

Reply:
+1

Reply:
Tx hashes from outport are hex encoded.
When I filter them, I need to fill mbs with `[][]byte` hashes for miniblocks.

I need to decode the hex strings in order to fill them in `&block.MiniBlock` when using `createSortedMbs`, so I need those.

Reply:
ok